package.json

```diff
@@ -110,6 +110,7 @@
     "socket.io-client": "^3.1.3",
     "sweetalert2": "^10.15.6",
     "tslib": "^2.1.0",
+    "url-pattern": "^1.0.3",
```

Comment:
I would prefer that you use a widely used URL parser like parseurl. I would even prefer that you rely only on express URI parameters extraction but that seems difficult here. 

You can use https://www.npmtrends.com/url-pattern-vs-parseurl-vs-parseuri-vs-url-parse to compare npm package usage.

Reply:
No but go for that one then.